forch/forch_metrics.py

```diff
@@ -36,37 +36,43 @@ def stop(self):
         """Kill varz server"""
         self._http_server.stop_server()
 
-    def _get_varz(self, var):
+    def _get_varz(self, var, labels=None):
         varz = self._metrics.get(var)
+        if labels:
+            varz = varz.labels(*labels)
         if not varz:
             LOGGER.error('Error updating to varz %s since it is not known.', var)
             raise RuntimeError('Unknown varz')
         return varz
 
-    def update_var(self, var, value):
+    def update_var(self, var, value, labels=None):
         """Update given varz with new value"""
-        varz = self._get_varz(var)
+        varz = self._get_varz(var, labels)
         if isinstance(varz, Info):
             varz.info(value)
+        elif isinstance(varz, Gauge):
+            varz.set(value)
         else:
             error_str = 'Error incrementing varz %s since it\'s type %s is not known.' \
                 % (var, type(varz))
             raise RuntimeError(error_str)
 
-    def inc_var(self, var, value=1):
+    def inc_var(self, var, value=1, labels=None):
         """Increment Counter or Gauge variables"""
-        varz = self._get_varz(var)
-        # TODO: If a Gauge var is ever added, update to check for Gauge type
-        if isinstance(varz, Counter):
+        varz = self._get_varz(var, labels)
+        if isinstance(varz, (Counter, Gauge)):
             varz.inc(value)
         else:
             error_str = 'Error incrementing varz %s since it\'s type %s is not known.' \
                 % (var, type(varz))
             raise RuntimeError(error_str)
 
-    def _add_var(self, var, var_help, metric_type):
+    def _add_var(self, var, var_help, metric_type, labels=None):
         """Add varz to be tracked"""
-        self._metrics[var] = metric_type(var, var_help, registry=self._reg)
+        if labels:
+            self._metrics[var] = metric_type(var, var_help, labels, registry=self._reg)
```

Comment:
is it possible to do labels=labels here?  Would be better if the if statement could be avoided.

Reply:
Removed if